ecommerce/frontend/src/components/SearchInput.js

```diff
@@ -14,35 +16,38 @@ const categories = [
     { key: 'womens_fashion', title: "Women's Fashion" },
 ]
 
-const { Option } = Select
+export const SearchInput = (width = 120, onSearch) => {
+    const [category, setCategory] = useState('Electronics')
 
-function handleCategoryChange(value) {
-    console.log(`selected ${value}`)
-}
+    function handleCategoryChange(value) {
+        setCategory(value)
+    }
 
-const categoryDropdown = (
-    <Select defaultValue="Electronics" style={{ width: 120 }} onChange={handleCategoryChange}>
-        {categories.map(category => (
-            <Option value={category.title}>
-                {category.title} key = {category.key}
-            </Option>
-        ))}
-    </Select>
-)
+    const categoryDropdown = (
+        <Select defaultValue={category} style={{ width: { width } }} onChange={handleCategoryChange}>
+            {categories.map(category => (
+                <Option value={category.title} key={category.key}>
+                    {category.title}
+                </Option>
+            ))}
+        </Select>
+    )
 
-const onSearchInput = value => {
-    if (value === '') {
-        return
-    }
-    // Trigger products page
-    console.log(value)
-}
+    const onSearchInput = value => {}
 
-export const SearchInput = () => {
     return (
         <div className="search-input">
             {categoryDropdown}
-            <Search placeholder="Ürün, kategori veya marka ara" onSearch={onSearchInput} enterButton />
+            <Search
+                placeholder="Search for products, categories and brands"
+                onSearch={value => {
+                    if (value.trim() === '') {
+                        return
+                    }
+                    onSearch(value)
+                }}
```

Comment:
Might as well write `onSearch(value.trim())` so that any whitespace is removed too.